api/controllers/analytics.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const request = require('request');
+const config = require('../config');
+const common = require('../common');
+
+const LabelNameMatch = {
+    "PhD Student": ["doctoral student", "graduate student","phd candidate","grad student","phd student"],
+    "Faculty" : ["prof", "senior research fellow", "chair", "instructor", "director", "scientist", "advisor", "principal investigator", "pi", "teacher","scholar", "lec"],
+    "Postdoctoral Researcher" : ["postdoctoral" ,"fellow", "research associate", "post", "phd" ,"research"],
+    "Research Assistant" : ["research assistant", "research coordinator", "intern","ra"],
+    "High School Student" : ["high school"],
+    "Clinician" : ["chief physician","clinical researcher","advanced imaging neuroscientist","presurgical tumor evaluation","md","radiologist","neuroradiologist",
+                "logist","physic","clinician","medical doctor","physician","neurologist"],
+    "College Student" : ["under", "teaching assistant",],
+    "Industry" : ["software", "product", "manager", "owner", "developer", "des", "engineer"],
+    "Masters Student" : ["masters", "phil", "mtech", "msc"],
+    "Student (unspecified)" : ["studen"],
+}
+/**
+ * @apiGroup Projects
+ * @api {get} /project/data
+ * @apiDescription              Query data for pie chart
+ * 
+ * @apiHeader {String} [authorization]
+ *                                  A valid JWT token "Bearer: xxxxx"
+ * @apiSuccess {Object (JSON) }             Pie Chart Data
+ */
+ router.get('/data', common.jwt({credentialsRequired: true}), (req, res, next)=> {
+    // if(!req.headers.authorization) next();
+    if(!common.hasScope(req, "admin")) return next("You need to be admin to access this");
+    request.get({ url : config.auth.api+'/profile/list', headers: { authorization: req.headers.authorization}}, (err, _res, json)=> {
+        if(err) return next(err);
+        if(json.length == 0) return res.json([]); 
+        //now we have the json, we need to filter it
+        let dataFiltered = JSON.parse(json).profiles.map(entry => {
+            if(entry.profile.private.position) return {position : entry.profile.private.position};
+        });
+        if(!dataFiltered.length) return res.json([]);
+        return res.send(LabeltoValue(dataFiltered,LabelNameMatch));
+    });
+});
+
+function LabeltoValue(data,LabelNameMatch) {
```

Comment:
I don't think we need a separate function for this. It's too specific to what the API is doing, and it's not re-used anywhere else (also too complex)